### PR TITLE
[AFTERSHOCK] Basic Ballistic Weapons

### PR DESCRIPTION
**Pull request by Mom-Bun**
#### Summary
Mods "Adding Basic Ballistic Weapons to Aftershock"

#### Purpose of change

Considering that Aftershock is set 500 years in the future, many of the guns currently in game wouldn't make much sense being used, and in general it is hard to make a general feel for combat balance with vanilla weapons and drastically different enemies.

This PR should add some 3 basic ammo typos and 4 guns to Aftershock that'll help provide stepping stones towards this, as well fill out the combat balance document for the AFS wiki. 

#### Describe the solution

I added 4 basic weapons that fall under the umbrella of pistol, shotgun, rifle, and machine gun; and ammo types that should go along with these.

### Gibson S86
This light grenade launcher should fulfill the criteria of being a heavy weapon for the combat balance document, it should use the 25mm ammo.

### Seyfert 84K
This pistol should fulfill the criteria of being a light weapon for the combat balance document. It uses the 10mm ammo.

### G&W SR-P9
This rifle should fulfill the criteria of being a medium weapon for the combat balance document. It uses the 6.5mm ammo.

### Accipiter Hawk-00
This machine gun should fulfill the criteria of being a medium weapon for the combat balance document. It uses the 6.5mm ammo.
 
## Ammo

One of the ideas in Aftershock is we want to both avoid the highly varied amount of ammo by using the same idea as generic guns ammo. The reason for this is both in-universe flavor, and game play reasons. Ammo in the Aftershock universe is very highly standardized and unified and one of the most common complaints is that the highly varied ammo in base game Cataclysm is very confusing.

### 25mm
This type of ammo should be saved for light grenade launchers, in general very hard hitting, but not the top tier weapons. Heavy grenade launchers should not use this.

These are not powerful enough to actually make a substantial explosion except to damage a mob.

### 10mm
This is the basic ammo for all modern earth-made pistols, and every pistol that is modern and earth-made, unless it is high tier or specialized pistols should use this type of ammo.

### 7.50mm
This is the basic ammo for all modern earth-made rifles and machine guns, the idea being that they are an intermediary bullet. These should be used with every rifle and machine gun that is modern and earth-made, unless it is exceptionally special or high-tier to call for unique ammo. 

TBD:

~- [x] Add gun bases for the rifle, machine gun, and pistol~ Unnecessary 
- [x] Fill in description and add flavor
- [x] Add JHP variant of 10mm
- [x] Add turret mount for the Hawk and s140
- [x] Test balance to make sure they fall in the ranges that is needed for the combat balance
- [x] Add ammo and guns to gun store drops.

#### Describe alternatives you've considered

Still using the same ammo and guns as in base Cataclysm.

#### Testing

Game boots up, guns work in the intended ranges I wanted them to; no errors and items drop.

I welcome that if other the other aftershock maintainers want to test the balance for the guns, it would be greatly appreciated.

#### Additional context

To note this isn't exhaustive as a PR, and merely here to help fill in the combat balance document by filling in ballistic weapons for Aftershock. For reference in the future we'd need ammo for shotguns, regular grenade launchers, and sub machine guns. 

Another note is not all ammo is caseless, in the future there can be variants of ammo that can have cases; but they should have reason for this, either because they were handloaded by PrepNet Phyle, or it is very extraordinary/high tier as a bullet.

### CODE REVIEW

File: data/mods/Aftershock/items/ammo/ballistic.json
Comment (actual-nh) [inline data/mods/Aftershock/items/ammo/ballistic.json:6]:
```suggestion
    "description": "25x40mm caseless rounds.  The most common type of ammo used for high-gauge shotguns and light grenade launchers.  Being caseless, these cannot be disassembled or reloaded.",
```

File: data/mods/Aftershock/items/ammo/ballistic.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/ammo/ballistic.json:55]: This flag means the item becomes more expensive as the game calendar progresses, to imply the DDA post-apocalypse cant produce more. Not fully sure you intended to add it?

Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/ballistic.json:55]: Ahhh didn't know that part; Let me remove it.

File: data/mods/Aftershock/items/ammo/10mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/10mm.json:6]:
```suggestion
    "description": "10mm ammunition with a brass jacketed bullet.  For as long as you can remember bullets like these have been used for military, law enforcement, and civilian use.  Being caseless, these cannot be disassembled or reloaded.",
```

File: data/mods/Aftershock/items/ammo/10mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/10mm.json:11]:
```suggestion
    "material": [ "brass" ],
```
We can ignore whatever combustible adhesive we are using in place of powder for now?  Or we can create the material and allow it to be scraped off and turned into makeshift explosives?

File: data/mods/Aftershock/items/ammo/10mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/10mm.json:32]:
```suggestion
    "material": [ "brass" ],
```
Same as above

File: data/mods/Aftershock/items/ammo/25mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/25mm.json:11]:
```suggestion
    "material": [ "brass" ],
```
Same as above

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/6.5mm.json:11]:
```suggestion
    "material": [ "brass" ],
```
Same as above

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/ammo/6.5mm.json:32]:
```suggestion
    "material": [  "lead" ],
```
Maybe we need to actually remove the brass above and replace them with lead and in heavy duty rounds DUP (depleted Uranium)

File: data/mods/Aftershock/items/magazine/25mm.json
Comment (Maleclypse) [inline data/mods/Aftershock/items/magazine/25mm.json:12]: This is the only gun or magazine made using superalloy?  I feel like we should either use more superalloy or not use it here.

**Review by Maleclypse — COMMENTED** [review]
I'm not sure what direction we should take on some of these items but I wanted us to come up with a consistent application.

File: data/mods/Aftershock/items/gun/25mm.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/25mm.json:22]: This burst is probably too long. Havent play tested it but wouldnt the last 3 tot 2 shots just completely miss almost allways?

File: data/mods/Aftershock/items/gun/10mm.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/10mm.json:8]:
```suggestion
    "description": "Commonly carried by shipping crews that can afford it, Orion Armistice's 84K pistol doesn't particulary shine or show anything special but it keeps you safe on long journeys in the vast dark of Orion's Arm.",
```
Edited to describe it as a pistol, and to remove some repetitions of "Orion".
Aditionally, I would consider a different manufacturer to avoid repeating Orion at all.

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/ammo/6.5mm.json:27]: These aren't caseless?

They probably should be as the other 6.5 guns can fire them alright.

File: data/mods/Aftershock/items/gun/25mm.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/25mm.json:8]: Is it really a shotgun? My understanding was that it was a grenade launcher, and you can happen to load shot-equivalent shells into it.

Mostly say this because real 40mm grenade launchers have shells that are basically oversized 00 shot [the one that says canister in here ](https://en.wikipedia.org/wiki/40_mm_grenade). But having guns designed shotguns shoot grenades seems weirder

File: data/mods/Aftershock/items/ammo/10mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/10mm.json:11]: Probably best to ignore it for now; and in the future maybe allow it to be scraped off?

File: data/mods/Aftershock/items/magazine/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/magazine/25mm.json:12]: Ahhh, yeah more of the magazines should use it!

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/6.5mm.json:27]: Ah yes, they are caseless

File: data/mods/Aftershock/items/gun/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/25mm.json:22]: Correct yeah; most of em miss

The idea was it is very inaccurate, however if this is a problem I can lower it.

File: data/mods/Aftershock/items/ammo/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/25mm.json:5]: Can remove the comma yeah, also no idea there; I'm pretty sure 40mm thing is the size of dimensions, so its 25 mm wide, but 40mm in length? 

I can remove it however for consistencies sake.

And yeah; I imagine any explosions made from these won't be enough to really demolish things, just enough to cause damage to monsters.

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/6.5mm.json:32]: Agreed

File: data/mods/Aftershock/items/gun/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/25mm.json:8]: Ah that makes sense; can re-write the flavor for this then!

File: data/mods/Aftershock/items/ammo/25mm.json
Comment (actual-nh) [reply data/mods/Aftershock/items/ammo/25mm.json]: I agree re not caseless; if it's got a canister already...

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (John-Candlebury) [reply data/mods/Aftershock/items/ammo/6.5mm.json]: I 'm not sure about that switch. It seems likely that the design considerations to make an hybrid gun like that would have serious to severe drawbacks. So it likely wouldnt happen even with very good tech.

Mostly I say this because I'm reasonably sure that players that do like guns are going to complain about this.

File: data/mods/Aftershock/items/gun/25mm.json
Comment (John-Candlebury) [reply data/mods/Aftershock/items/gun/25mm.json:22]: Dont think balancing a heavy weapon around most shots missing is exactly right.  A precise long burst is fine because it easily kills strong enemies and wastes a lot of  ammo overkilling weak foes. But a non precise burst isnt really good at anything but wasting your resources, and would likely mean that the gun will go unused.

A high miss percentage is fine if you dont have skills and randomly pick up the gun while having the mininum requirements. But someone with the adequate strenght boosts + skills should probably be able to land the whole burst against large oponents.

Comment (John-Candlebury) [reply data/mods/Aftershock/items/gun/25mm.json:22]:
```suggestion
    "modes": [ [ "DEFAULT", "multi", 6 ] ],
```
This might allow the 6 bullets to hit without needing many overreaching changes. It would have to be tested however.

File: data/mods/Aftershock/items/ammo/6.5mm.json
Comment (actual-nh) [reply data/mods/Aftershock/items/ammo/6.5mm.json]: I suspect that it would be possible to convert the weapons, but it wouldn't be as simple as flipping a switch - replacing the loading mechanism at minimum.

File: data/mods/Aftershock/items/gun/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/25mm.json:22]: Can do; I'll try testing it out later

Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/25mm.json]: I ended up lowering it down to 4 and setting it to DOUBLE, as no matter how I fiddle with it; every shot missed until I lowered it down.

**Comment by actual-nh** [comment]
AFS wiki?

**Comment by Mom-Bun** [comment]
> AFS wiki?

Something me and the other maintainers have been working on; once its polished and ready we will try to get it on the main repo

**Comment by actual-nh** [comment]
> > AFS wiki?
> 
> Something me and the other maintainers have been working on; once its polished and ready we will try to get it on the main repo

I have been doing some work on the overall Cataclysm wiki - it could use (much more) [material](http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Aftershock) on Aftershock.

File: data/mods/Aftershock/items/gun/7.50mm.json
Comment (John-Candlebury) [inline data/mods/Aftershock/items/gun/7.50mm.json:75]:
```suggestion
    "description": "An eminently sturdy and reliable  bullpop rifle iused by UICA's military during early reclamation expeditions.  Although its been out of service for nearly two decades, many a private military and contractor has found a use for the SR-P9, and it remains common long past its initial production date.",
```
There was something that sounded wrong about the last part of the last sentence, but Im not necesarily sure that my suggestion for it is much better.

File: data/mods/Aftershock/items/casing.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/casing.json:7]: I took your suggestion to make them cased, yes

File: data/mods/Aftershock/items/ammo/25mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/ammo/25mm.json:10]: Of note I just went through all items in this PR, and the alien battery, but it might be worth while to go looking over items in aftershock and due an audit in the future to remove the post_apoc prices.

File: data/mods/Aftershock/items/gun/7.50mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/7.50mm.json:27]: It is still intended to be usable, not sub-par yes.

File: data/mods/Aftershock/items/gun/7.50mm.json
Comment (Mom-Bun) [reply data/mods/Aftershock/items/gun/7.50mm.json:75]: I think maybe the "production date" is the issue here, so I'll probably change that to "initial introduction into service"

**Comment by Mom-Bun** [comment]
> > > AFS wiki?
> > 
> > 
> > Something me and the other maintainers have been working on; once its polished and ready we will try to get it on the main repo
> 
> I have been doing some work on the overall Cataclysm wiki - it could use (much more) [material](http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Aftershock) on Aftershock.

Ah I haven't touched that, I wasn't even aware there was a wiki page for Aftershock on the wiki, I was referring to a github repo wiki

This could use some touching up in the future possibly.

**Review by John-Candlebury — APPROVED** [review]
Probably not going to make it for 0.F,  but this one is good to go once freeze is lifted

**Pull request merged**
